### PR TITLE
Cleanup of a couple of `TODO` statements

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
This PR cleans up a couple of `TODO` statements in our code:

1. Drop the `kube-apiserver-kubelet` Helm chart part of `shoot-core` (refactored with https://github.com/gardener/gardener/pull/4534, released with `v1.31`)
1. Drop the `coredns` Helm chart part of `shoot-core` (refactored with https://github.com/gardener/gardener/pull/4536, released with `v1.32`)
1. Drop the `vpn-shoot` Helm chart part of `shoot-core` (refactored with https://github.com/gardener/gardener/pull/5056, released with `v1.38`)
1. Drop the `node-local-dns` Helm chart part of `shoot-core` (refactored with https://github.com/gardener/gardener/pull/5122, released with `v1.39`)
1. Drop the `node-problem-detector` Helm chart part of `shoot-core` (refactored with https://github.com/gardener/gardener/pull/5204, released with `v1.39`)
1. Drop the `kube-proxy` Helm chart part of `shoot-core` (refactored with https://github.com/gardener/gardener/pull/5438, released with `v1.41`)
1. Remove `security.gardener.cloud/trigger` annotations (added with https://github.com/gardener/gardener/pull/5099 and https://github.com/gardener/gardener/pull/5128, released with `v1.38`)
1. Remove special handling for ambiguous gardenlets (support dropped with https://github.com/gardener/gardener/pull/5093, released with `v1.38`)
1. Remove migration coding when eliminating client certificates (most of these changes were released prior to `v1.38`)
1. Remove migration coding for eliminating duplicate project members (added with https://github.com/gardener/gardener/pull/3137, released with `v1.13`)
1. Drop support for `terraformer` v1 (v2 is the default since https://github.com/gardener/gardener/pull/4930, released with `v1.36`)

**Release note**:

```breaking developer
Support for Terraformer `v1` was finally dropped.
```

### CODE REVIEW

**Review by acumino — COMMENTED** [review]
Can you adapt changes for terraformer here also https://github.com/gardener/gardener/blob/c78fb79c26833f406abaf9a632e0872e9796a339/extensions/pkg/terraformer/terraformer.go#L400

File: extensions/pkg/terraformer/terraformer.go [hunk 1]
```diff
@@ -427,19 +421,7 @@ func getTerraformerCommand(pod *corev1.Pod) string {
 }
 
 func (t *terraformer) env() []corev1.EnvVar {
```
Comment (acumino) [inline extensions/pkg/terraformer/terraformer.go]: Do we still need this function? Probably we should drop it.

**Comment by acumino** [comment]
Thanks
/lgtm

**Comment by ary1992** [comment]
> 5. Drop the node-problem-detector Helm chart part of shoot-core (refactored with https://github.com/gardener/gardener/pull/5204, released with v1.39)

/lgtm

**Review by timebertt — COMMENTED** [review]
Thanks!
/lgtm
/approve

**Comment by rfranzke** [comment]
Nice @acumino !

**Pull request merged**